### PR TITLE
internal/v5: fix revision-info

### PR DESCRIPTION
**Pull request by rogpeppe**
The old version did not respect the fact that the development
channel was not stored in the entity URL.
Also we now respect permissions in revision-info
and fix a bug in entitycache that could cause
entities to be returned out of order in an iterator.

### CODE REVIEW

**Comment by mhilton** [comment]
:+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1486/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1487/
Test PASSed.

File: internal/v5/api.go [hunk 1]
```diff
@@ -771,25 +771,29 @@ func (h *ReqHandler) metaRevisionInfo(id *router.ResolvedURL, path string, flags
...
+	for iter.Next() {
+		e := iter.Entity()
+		rurl := charmstore.EntityResolvedURL(e)
+		if err := h.AuthorizeEntity(rurl, req); err != nil {
+			// We're not authorized to see the entity, so leave it out.
+			// Note that the only time this will happen is when
+			// the original URL is promulgated and has a development channel,
+			// the charm has changed owners, and the old owner and
+			// the new one have different dev ACLs. It's easiest
+			// and most reliable just to check everything though.
```
Comment (frankban) [inline internal/v5/api.go:785]: +1

**Comment by frankban** [comment]
Great branch Roger, thanks!
:+1:

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**